x-pack/platform/plugins/shared/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -1102,32 +1100,24 @@ export const LensTopNavMenu = ({
     [editField, canEditDataView]
   );
 
-  const createNewDataView = useCallback(() => {
-    closeDataViewEditor.current = dataViewEditor.openEditor({
-      onSave: async (dataView) => {
-        if (dataView.id) {
-          if (isOnTextBasedMode) {
-            dispatch(
-              switchAndCleanDatasource({
-                newDatasourceId: 'formBased',
-                visualizationId: visualization?.activeId,
-                currentIndexPatternId: dataView?.id,
-              })
-            );
-          }
-          dispatchChangeIndexPattern(dataView);
-          setCurrentIndexPattern(dataView);
+  const createNewDataView = useCallback(
+    async (dataView: DataView) => {
+      if (dataView.id) {
+        if (isOnTextBasedMode) {
```

Comment:
Can you please explain why there is no need for the `dataViewEditor.openEditor` call here?
I can't follow all the changes because they belongs to a codebase knowledge I don't have, but it would be great to know why.
Reading the PR description I see:
>As the variation of the editor form increase from 2 scenarios to 4, additional logic is added. A small refactoring was made to onDataViewCreated to standardized the props, so that it can be reused by the duplicate logic (when duplicating, it is essentially the same behavior as creating a data view.

But this doesn't give me enough info to understand why I don't need anymore to "open the editor first", is it already open? if so can you point me on who is opening the editor now? thanks

Reply:
Perfect thanks for the details